### ISSUE TITLE
Question regarding carrier-grade NAT

### ISSUE THREAD

**Issue by salokinsomisa** [lontivero/Open.NAT#92]
Hello, I'd like to ask if support for Carrier-grade NAT and PCP are included in this implementation. I'm asking this because I've been trying to use Open.NAT to port-forward a device that I own and the external IP I'm getting from Open.NAT is different from the one I'm getting if I search for my IP in a browser. The uPnP rule is added to my router, but the port seems closed ( It probably has to do with the two IPs being different, which prompted me to ask for Carrier-grade NAT).

Any ideas on how to counter this? (Perhaps the ISP does something different than carrier-grade NAT, but it's 100% sure that the WAN IP the device has is different than the public IP the world is seeing)

**Comment by lontivero** [comment]
No sure I understand the question. Are you asking if you can open a port in somebody else's routing infrastructure as your ISP or your ISP's ISP? In that case the answer is: no, you can only operate on your own infrastructure components.

**Comment by salokinsomisa** [comment]
I just found that in some cases, the public IP and the WAN IP are different. The WAN IP of my computer is a private one and I read that Carrier-grade NAT is probably behind. Though port-forwarding works, the computer is not accessible through the internet since it's behind a private WAN. I read that the PCP (Port Control Protocol) or its predecessor the NAT-PMP could solve this problem and I posted to ask if anyone knew about this problem.

In computers that don't hide behind a private WAN, everything works fine of course.
